kube-client/src/config/file_config.rs

```diff
@@ -358,26 +352,64 @@ fn to_absolute(dir: &Path, file: &str) -> Option<String> {
 }
 
 impl Cluster {
-    pub(crate) fn load_certificate_authority(&self) -> Result<Option<Vec<u8>>> {
+    pub(crate) fn load_certificate_authority(&self) -> Result<Option<Vec<u8>>, KubeconfigError> {
         if self.certificate_authority.is_none() && self.certificate_authority_data.is_none() {
             return Ok(None);
         }
-        let res =
-            utils::data_or_file_with_base64(&self.certificate_authority_data, &self.certificate_authority)?;
-        Ok(Some(res))
+
+        let ca = load_from_base64_or_file(&self.certificate_authority_data, &self.certificate_authority)
+            .map_err(KubeconfigError::LoadCertificateAuthority)?;
+        Ok(Some(ca))
     }
 }
 
 impl AuthInfo {
-    pub(crate) fn load_client_certificate(&self) -> Result<Vec<u8>> {
-        utils::data_or_file_with_base64(&self.client_certificate_data, &self.client_certificate)
+    pub(crate) fn load_client_certificate(&self) -> Result<Vec<u8>, KubeconfigError> {
+        load_from_base64_or_file(&self.client_certificate_data, &self.client_certificate)
+            .map_err(KubeconfigError::LoadClientCertificate)
     }
 
-    pub(crate) fn load_client_key(&self) -> Result<Vec<u8>> {
-        utils::data_or_file_with_base64(&self.client_key_data, &self.client_key)
+    pub(crate) fn load_client_key(&self) -> Result<Vec<u8>, KubeconfigError> {
+        load_from_base64_or_file(&self.client_key_data, &self.client_key)
+            .map_err(KubeconfigError::LoadClientKey)
     }
 }
 
+fn load_from_base64_or_file<P: AsRef<Path>>(
+    value: &Option<String>,
+    file: &Option<P>,
+) -> Result<Vec<u8>, LoadDataError> {
+    let data = value
+        .as_deref()
+        .map(load_from_base64)
+        .or_else(|| file.as_ref().map(load_from_file))
+        .unwrap_or(Err(LoadDataError::NoBase64DataOrFile))?;
+    Ok(ensure_trailing_newline(data))
+}
```

Comment:
The corresponding function looked like the following before this change:
```rust
fn data_or_file_with_base64<P: AsRef<Path>>(data: &Option<String>, file: &Option<P>) -> Result<Vec<u8>> {
    let mut blob = match (data, file) {
        (Some(d), _) => base64::decode(&d)
            .map_err(ConfigError::Base64Decode)
            .map_err(Error::Kubeconfig),
        (_, Some(f)) => read_file(f),
        _ => Err(Error::Kubeconfig(ConfigError::NoBase64FileOrData)),
    }?;
    //Ensure there is a trailing newline in the blob
    //Don't bother if the blob is empty
    if blob.last().map(|end| *end != b'\n').unwrap_or(false) {
        blob.push(b'\n');
    }
    Ok(blob)
}
```

I changed the error to a separate `LoadDataError`, so we can describe errors like "decoding base64 failed when loading client key" instead of "decoding base64 failed in kubeconfig".

Changed to `.map`, `.or_else`, `.unwrap_or(err)` because I think it makes the intent more obvious.